unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -159,6 +162,77 @@ public UTF8String toLowerCase() {
     return UTF8String.fromString(toString().toLowerCase());
   }
 
+  public UTF8String trim() {
+    return UTF8String.fromString(toString().trim());
```

Comment:
my understanding is that in a well-formed utf-8 byte sequence, the last byte in a code point cannot be 0x20, unless it is a space character.